app/src/main/java/com/hover/stax/home/MainActivity.java

```diff
@@ -89,23 +98,48 @@ else if(deepLinkRoute.contains(getString(R.string.deeplink_settings))) {
                 navigateToSettingsFragment(getNavController());
             }
             else if(deepLinkRoute.contains(getString(R.string.deeplink_reviews))) {
-                Utils.logAnalyticsEvent(getString(R.string.visited_rating_review_screen), this);
-                if(!Utils.getBoolean(Constants.APP_RATED, this)) launchRatingAndReviewDialog();
-                else openStaxPlaystorePage();
+                Timber.i("Currently at reviews 1");
+                launchStaxReview();
             }
         }
     }
+    private void observeTransactionsForAppReview() {
+        transactionHistoryViewModel.getStaxTransactionsForAppReview().observe(this, staxTransactions -> {
+                if(qualifiesForAppReview(staxTransactions)) launchRatingAndReviewDialog();
+        });
+    }
+    private boolean qualifiesForAppReview(List<StaxTransaction> staxTransactions) {
```

Comment:
Let's have this happen in the `ViewModel` and only post the result to the fragment. It's better for performance.

Reply:
Thanks for this, refractor done.